messaging/src/main/java/org/axonframework/eventhandling/EventTrackerStatus.java

```diff
@@ -49,6 +51,21 @@
      */
     boolean isReplaying();
 
+    /**
+     * Indicates whether this Segment is still merging events.
+     *
+     * @return {@code true} if this segment is merging events, otherwise {@code false}
+     */
+    boolean isMerging();
+
+    /**
+     * Return the estimated relative token position this Segment will have after a merge operation is complete.
+     * In case no estimation can be given or no merge in progress, an {@code OptionalLong.empty()} will be returned.
+     *
+     * @return return the estimated relative position this Segment will reach after a merge operation is complete.
+     */
+    OptionalLong mergeCompletedPosition();
```

Comment:
I have a naming suggestion for this method. What about we call it the `mergingFinishPosition` or ``mergingGoalPosition`?

I was thinking about what this field represent, and that's essentially the "finish line when merging will be resolved".

Reply:
After a quick round of consultations, the team decided to stick with my proposal mergeCompletedPosition()